src/quicklist.c

```diff
@@ -924,11 +1020,17 @@ REDIS_STATIC void _quicklistInsert(quicklist *quicklist, quicklistEntry *entry,
         new_node->count++;
         quicklistNodeUpdateSz(new_node);
         quicklistRecompressOnly(quicklist, new_node);
-    } else if (full && ((at_tail && node->next && full_next && after) ||
-                        (at_head && node->prev && full_prev && !after))) {
+    } else if (full && (((at_tail && node->next && full_next && after) ||
+                        (at_head && node->prev && full_prev && !after)) ||
+                        (node->container == QUICKLIST_NODE_CONTAINER_NONE))) {
```

Comment:
@oranagra Why not add at the start of `_quicklistInsert`.
```
if (sz >= packed_threshold || node->container == none) { 
    create new node 
}
```

Reply:
@sundb there are cases where the string is small and can fit into the beginning of the next node instead of the tail of the current one, and alike. your suggestion will prevent that, right?